unified-runtime/source/adapters/cuda/event.hpp

```diff
@@ -16,39 +16,28 @@
 #include "queue.hpp"
 
 /// UR Event mapping to CUevent
-///
 struct ur_event_handle_t_ : ur::cuda::handle_base {
-public:
   using native_type = CUevent;
 
-  ur_result_t record();
-
-  ur_result_t wait();
+  ur_event_handle_t_(
+      ur_command_t Type, ur_queue_handle_t Queue, CUstream Stream,
+      uint32_t StreamToken = std::numeric_limits<uint32_t>::max());
+  ur_event_handle_t_(ur_context_handle_t Context, CUevent EventNative);
```

Comment:
Not sure I understand this constructor, it looks like a new interface?
This constructor only has a ur context and a CUevent. Isn't an event always associated with a queue?

Reply:
It's for interop, it's used in `urEventCreateWithNativeHandle`, these interop events don't have an associated queue

Reply:
I'm not sure either, I don't know if the event interop is actually used, but that's essentially the signature of `urEventCreateWithNativeHandle`.

This was already there before just with an extra level through `makeWithNative`.